src/server/lang.js

```diff
@@ -0,0 +1,43 @@
+import path from "path"
+import {readdir} from "fs"
+
+import {addLocaleData} from "react-intl"
+import fr from "react-intl/locale-data/fr"
+import en from "react-intl/locale-data/en"
+
+const DEFAULT_LANGUAGE = 'fr'
+
+export function loadTranslations(langDir){
+  addLocaleData(...fr)
+  addLocaleData(...en)
+
+  var messages = new Map()
+  var dotIndex = -1
+  var json
+  readdir(langDir, (err, files) => { 
+    if(err){
+      console.log("Unable to load translation files.", err)
+    }
+
+    files.forEach(file => {
+      dotIndex = file.indexOf('.')
+      json = path.resolve(langDir, file)
+      messages.set(file.substring(0, dotIndex), require(json))
+    })
+  })
+  return messages
+}
+
+export function getLocale(acceptLanguage, messages){
+  var locale = acceptLanguage ? acceptLanguage.substring(0, 2) : DEFAULT_LANGUAGE //ex: en-US;q=0.4,fr-FR;q=0.2 > en
```

Comment:
This is not correct: you only accept a match on the first user preference. Do not reinvent parsing at all and add more functionality by using a package such as https://github.com/tinganho/node-accept-language.